### PR TITLE
Use unique port names in Gateway servers

### PR DESCRIPTION
**Pull request by rshriram**
Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

### CODE REVIEW

**Comment by baodongli** [comment]
one of the failures in ci/circleci seems to be caused by the hard-coded port name found in this file pilot/pkg/config/kube/ingress/conversion.go

File: tests/testdata/config/virtual-service-all.yaml [hunk 1]
```diff
@@ -6,8 +6,7 @@ spec:
   hosts:
   - service3.default.svc.cluster.local
   - uk.bookinfo.com
-  # Bug: dup not rejected
-  #- eu1.bookinfo.com
+  - eu1.bookinfo.com
```
Comment (baodongli) [inline tests/testdata/config/virtual-service-all.yaml]: the same host is defined in rule-route-via-egressgateway.yaml

Comment (rshriram) [reply tests/testdata/config/virtual-service-all.yaml]: Ah thats why its failing..

File: istioctl/cmd/istioctl/convert/testdata/v1alpha1/myservice-ingress.yaml [hunk 2]
```diff
@@ -4,6 +4,7 @@ metadata:
   annotations:
     kubernetes.io/ingress.class: istio
   name: simple-ingress
+  namespace: default
```
Comment (baodongli) [inline istioctl/cmd/istioctl/convert/testdata/v1alpha1/myservice-ingress.yaml]: Just a question: if the namespace is default, is it required to specify that as well?

File: istioctl/cmd/istioctl/convert/testdata/v1alpha3/merged-gateway.yaml.golden [hunk 3]
```diff
@@ -38,7 +38,7 @@ spec:
         prefix: /foo/
     route:
     - destination:
-        host: myservice-service
+        host: myservice-service.default.svc.cluster.local
```
Comment (baodongli) [inline istioctl/cmd/istioctl/convert/testdata/v1alpha3/merged-gateway.yaml.golden]: same question here. Thought it's not required to specify the full name in default namespace

File: istioctl/cmd/istioctl/convert/testdata/v1alpha1/myservice-ingress.yaml (hunk 2 above)
Comment (rshriram) [reply istioctl/cmd/istioctl/convert/testdata/v1alpha1/myservice-ingress.yaml]: it is not when you submit to kubernetes. this is going to the conversion tool.

File: istioctl/cmd/istioctl/convert/testdata/v1alpha3/merged-gateway.yaml.golden (hunk 3 above)
Comment (rshriram) [reply istioctl/cmd/istioctl/convert/testdata/v1alpha3/merged-gateway.yaml.golden]: this is the generated route rule. Using FQDN maintains clarity. The short name stuff has proven to be very ambiguous.

**Pull request merged**